gp_services.R

```diff
@@ -7,54 +7,66 @@
 
-#Read the JSON text from the following URL
-services <- fromJSON("https://ual.geoplatform.gov/api/communities/4eebc494059beab9fda54cb078927ddc/items?type=regp:Service&size=200")
+#Read the JSON text from the URL with the Resilience Community service information
+services <- fromJSON(ResilienceUrl)
 
 #Create an empty vector to hold all of the service IDs
 id <- c()
 title <- c() 
 
 #Loop through each service entry in the JSON file (now an R list)
+#to retrieve desired information, must use the index 'services[[1]]'
 for(i in 1:length(services[[1]])){
-  id[i] <- services[[1]][[i]][['id']] 
-  title[i] <- services[[1]][[i]][['label']] 
+  element <- services[[1]][[i]]
+  id[i] <- element[['id']]
+  title[i] <- element[['label']]
 }
 
 #Create an empty vector to hold all of the service urls
 serviceUrl <- c()
-service_types <- c()
+serviceTypes <- c()
 
 #Now loop though each service to get its URL (this could take a few seconds)
 for(i in 1:length(services[[1]])){
-  service_info <- fromJSON(paste0("https://ual.geoplatform.gov/api/services/", id[i]))
-  serviceUrl[i] <- service_info$href
-  service_types[i] <- service_info$serviceType$label
+  serviceInfo <- fromJSON(paste0("https://ual.geoplatform.gov/api/services/", id[i]))
+  serviceUrl[i] <- serviceInfo$href
+  serviceTypes[i] <- serviceInfo$serviceType$label
 }
 #Substitute titles with FGDC required fields into a new vector 
-service_typesnew <- gsub("Esri REST Map Service", "agsmapserver", service_types)
+serviceTypesNew <- gsub("Esri REST Map Service", "agsmapserver", serviceTypes)
 
 #repeat step with replace service_types with service_typesnew until all replaced
-service_typesnew1 <- gsub("Esri REST Tile Service", "agsmapserver", service_typesnew)
-service_typesnew2 <- gsub("Esri REST Feature Service", "agsfeatureserver", service_typesnew1)
-service_typesnew3 <- gsub("Esri REST Image Service", "agsimageserver", service_typesnew2)
-service_typesnew4 <- gsub("OGC Web Map Service \\(WMS\\)", "wms", service_typesnew3)
-serviceType <- gsub("OGC Web Feature Service \\(WFS\\)", "wfs", service_typesnew4)
+serviceTypesNew1 <- gsub("Esri REST Tile Service", "agsmapserver", serviceTypesNew)
+serviceTypesNew2 <- gsub("Esri REST Feature Service", "agsfeatureserver", serviceTypesNew1)
+serviceTypesNew3 <- gsub("Esri REST Image Service", "agsimageserver", serviceTypesNew2)
+serviceTypesNew4 <- gsub("OGC Web Map Service \\(WMS\\)", "wms", serviceTypesNew3)
+serviceType <- gsub("OGC Web Feature Service \\(WFS\\)", "wfs", serviceTypesNew4)
 
 #Create a data frame of the info so far
-all_service_info <- data.frame(id, title, serviceType, serviceUrl, stringsAsFactors = FALSE)
-View(all_service_info) #View the data frame
+allServiceInfo <- data.frame(id, title, serviceType, serviceUrl, stringsAsFactors = FALSE)
+View(allServiceInfo) #View the data frame
+
+#Create a constant for the csv file
+CSVfile = "all_service_info.csv"

```

Comment:
Move this towards the top of the file, all constants together. (arrange constants in alphabetical order)